File: Backend/SOS-Clean-House-back/src/main/java/com/SOS_Clean_House_back/model/AtividadesDesenvolvidas.java
package com.SOS_Clean_House_back.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@AllArgsConstructor @NoArgsConstructor @Data @Builder
public class AtividadesDesenvolvidas {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private  Integer id;
    private List<String> atividades;
    private String valorDiario4H;
    private String valorDiario8H;
    private String tempoReserva;
    private List<String> diasDaSemana;
    private String servico;


    @ManyToOne
    @JoinColumn(name = "prestador_id")
    private Prestador prestador;



}

File: Frontend/SOS-Clean-House-front/src/app/componentes/interface/AtividadesDesenvolvidas.ts
import { Prestador } from "./Prestador";
import { Servico } from "./Servico";

export interface AtividadesDesenvolvidas {
    id: number;
    atividades: string[];
    valorDiario4H: string;
    valorDiario8H: string;
    tempoReserva: string;
    diasDaSemana: string[];
    servico: string;
  
    prestador: Prestador;
    servicos: Servico[];
  }